src/main/java/org/sagebionetworks/web/server/servlet/AppConfigServlet.java

```diff
@@ -116,21 +138,22 @@ public JSONObjectAdapter getLatestConfiguration() {
       if (!newConfigString.isEmpty()) {
         lastConfigValue = new JSONObjectAdapterImpl(newConfigString);
       }
+    } catch (BadRequestException | InternalServerException e) {
+      // Invalid token or server error, re-initialize the session to try to recover.
+      logger.log(
+        Level.SEVERE,
+        "Failed to get latest configuration, returning last or default configuration and attempting to re-initialize the session.",
+        e
+      );
+      initializeAppConfigClient();
+      return getLastConfigValueOrDefault();
```

Comment:
In the logs, a `BadRequestException` was thrown with message `"Token not valid"`. The tokens are one-time-use and my hypothesis is that we probably used a token more than once. Attempt to recover by re-initializing the session and getting a fresh token.